packages/grafana-data/src/types/plugin.ts

```diff
@@ -23,6 +23,12 @@ export enum PluginSignatureStatus {
   unsigned = 'unsigned', // no MANIFEST file
 }
 
+export enum PluginErrorCode {
+  invalid = 'invalid',
+  modified = 'modified',
+  unsigned = 'unsigned',
```

Comment:
as I think more... do we need different error codes for all the possible signature states?  Can we just have one that says "there is a problem with the signature" and then we look at the signature state (already available) for more details

Reply:
Yeah I had thought about that and had opted for more specificity at first as it allows us to better guide the user - but if we can afford to simplify that then that's great!

Reply:
Ah I see what you mean now. I didn't want to start having this state depend on other information in this first pass really, since this is introducing a design change to the plugin flow which may lead to weird side effects. Treating this error code as the source of truth allows us to plug something on top without interfering too much with what's already there.

But technically yeah we could just roll them up into one and see how that works out. Will wait to hear what the other folks think